AmplifyPlugins/Geo/AWSLocationGeoPlugin/AWSLocationGeoPlugin+ClientBehavior.swift

```diff
@@ -79,102 +109,171 @@ extension AWSLocationGeoPlugin {
-                             postalCode: $0.postalCode,
-                             country: $0.country)
+            return places
+        } catch {
+            let geoError = GeoErrorHelper.mapAWSLocationError(error)
+            throw geoError
         }
-
-        return .success(places)
     }
 
     // MARK: - Maps
 
     /// Retrieves metadata for available map resources.
     /// - Returns: Metadata for all available map resources.
-    public func availableMaps(completionHandler: @escaping Geo.ResultsHandler<[Geo.MapStyle]>) {
+    /// - Throws:
+    ///     `Geo.Error.accessDenied` if request authorization issue
+    ///     `Geo.Error.serviceError` if service is down/resource not found/throttling/validation error
+    ///     `Geo.Error.invalidConfiguration` if invalid configuration
+    ///     `Geo.Error.networkError` if request failed or network unavailable
+    ///     `Geo.Error.pluginError` if encapsulated error received by a dependent plugin
+    ///     `Geo.Error.unknown` if error is unknown
+    public func availableMaps() async throws -> [Geo.MapStyle] {
         let mapStyles = Array(pluginConfig.maps.values)
         guard !mapStyles.isEmpty else {
-            completionHandler(.failure(.invalidConfiguration(
+            throw Geo.Error.invalidConfiguration(
                 GeoPluginErrorConstants.missingMaps.errorDescription,
-                GeoPluginErrorConstants.missingMaps.recoverySuggestion)))
-            return
+                GeoPluginErrorConstants.missingMaps.recoverySuggestion)
         }
-        completionHandler(.success(mapStyles))
+        return mapStyles
     }
 
     /// Retrieves the default map resource.
     /// - Returns: Metadata for the default map resource.
-    public func defaultMap(completionHandler: @escaping Geo.ResultsHandler<Geo.MapStyle>) {
+    /// - Throws:
+    ///     `Geo.Error.accessDenied` if request authorization issue
+    ///     `Geo.Error.serviceError` if service is down/resource not found/throttling/validation error
+    ///     `Geo.Error.invalidConfiguration` if invalid configuration
+    ///     `Geo.Error.networkError` if request failed or network unavailable
+    ///     `Geo.Error.pluginError` if encapsulated error received by a dependent plugin
+    ///     `Geo.Error.unknown` if error is unknown
+    public func defaultMap() async throws -> Geo.MapStyle {
         guard let mapName = pluginConfig.defaultMap, let mapStyle = pluginConfig.maps[mapName] else {
-            completionHandler(.failure(.invalidConfiguration(
+            throw Geo.Error.invalidConfiguration(
                 GeoPluginErrorConstants.missingDefaultMap.errorDescription,
-                GeoPluginErrorConstants.missingDefaultMap.recoverySuggestion)))
-            return
+                GeoPluginErrorConstants.missingDefaultMap.recoverySuggestion)
+        }
+        return mapStyle
+    }
+    
+    @available(*, deprecated, message: """
```

Comment:
Are any of these deprecated methods needed at the plugin level?  Assuming the public completion handler APIs in the category all use the async/await methods in the plugin, recommend we remove all these from the plugin.